perun/collect/trace/run.py

```diff
@@ -1,206 +1,275 @@
+
+    :param Res res: the resources object
+    :param kwargs: the configuration settings for the collector
+    :returns: tuple (CollectStatus enum code,
                     string as a status message, mainly for error states,
-                    dict of kwargs and new values)
+                    dict of kwargs (possibly with some new values))
     """
-    log.cprint('Starting the post-processing phase... ', 'white')
-
-    # Get the trace log path
-    try:
-        resources = list(systemtap.trace_to_profile(**kwargs))
-
-        # Update the profile dictionary
-        kwargs['profile'] = {
-            'global': {
-                'timestamp': sum(res['amount'] for res in resources) / _MICRO_TO_SECONDS,
-                'resources': resources
-            }
+    WD.header('Post-processing phase... ')
+
+    # TODO: change the output according to the new format so that it doesn't use as much memory?
+    # Parse the records and create the profile
+    records = list(parse.trace_to_profile(res[Res.data()], **kwargs))
+    kwargs['profile'] = {
+        'global': {
+            'timestamp': sum(record['amount'] for record in records) / MICRO_TO_SECONDS,
+            'resources': records
         }
-        log.done()
-        result = _COLLECTOR_STATUS[systemtap.Status.OK]
-        return result[0], result[1], dict(kwargs)
+    }
 
-    except CalledProcessError as exception:
-        log.failed()
-        return _COLLECTOR_STATUS[systemtap.Status.EXCEPT], str(exception), dict(kwargs)
+    stdout.done('\n\n')
+    return CollectStatus.OK, "", dict(kwargs)
 
 
-def _validate_input(**kwargs):
-    """Validate the collector input parameters and transform them to expected format.
+def teardown(**kwargs):
+    """ Perform a cleanup of all the collection resources that need it, i.e. files, locks,
+    processes, kernel modules etc.
 
+    :param kwargs: the configuration settings for the collector
+    :returns: tuple (CollectStatus enum code,
+                    string as a status message, mainly for error states,
+                    dict of kwargs (possibly with some new values))
+    """
+    WD.header('Teardown phase...')
+
+    # Cleanup all the SystemTap related resources
+    if 'res' in kwargs:
+        systemtap.cleanup(**kwargs)
+    # Zip and delete the temporary and watchdog files
+    timestamp = kwargs.get('timestamp', time.strftime("%Y-%m-%d-%H-%M-%S", time.localtime()))
+    pid = kwargs.get('pid', os.getpid())
+    keep_temps = kwargs.get('keep_temps', False)
+    pack_name = os.path.join(
+        get_log_directory(), 'trace', 'collect_files_{}_{}.zip.lzma'.format(timestamp, pid)
+    )
+    with Zipper(kwargs.get('zip_temps', False), pack_name) as temp_pack:
+        if 'res' in kwargs:
+            _cleanup_collect_files(kwargs['res'], temp_pack, keep_temps)
+        WD.end_session(temp_pack)
+
+    stdout.done('\n\n')
+    return CollectStatus.OK, "", dict(kwargs)
+
+
+def _normalize_config(executable, **kwargs):
```

Comment:
Suggestion: Consider refactoring by extracting the config to isolate object.

Reply:
A good idea, I will rework it in the new tracer version where the configuration values changed a bit.